xbmc/music/MusicDatabase.cpp

```diff
@@ -4038,8 +4038,10 @@ bool CMusicDatabase::GetSongsFullByWhere(const std::string &baseDir, const Filte
     int total = -1;
 
     std::string strSQL = "SELECT %s FROM songview ";
-    if (artistData) // Get data from song and song_artist tables to fully populate songs with artists
-      strSQL = "SELECT %s FROM songview JOIN songartistview on songartistview.idsong = songview.idsong ";
+    if (artistData)
+      // Get data from song and song_artist tables to fully populate songs with artists. 
+      //Some songs may not have artists so Lfet join
```

Comment:
Yes it can really happen.
Bug was found by someome with exactly that in their database, real life data. Elsewhere JM used left joins, and he knew what he was doing. My oversight here.